### ISSUE TITLE
False positive in missing_package_linter for symbol

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1037]
```
lintr::lint("loadNamespace(package)\n", lintr::missing_package_linter())
```

lints:

```
<text>:1:1: warning: Package 'package' is not installed.
loadNamespace(package)
^~~~~~~~~~~~~~~~~~~~~~
```

As seen in our own code:

https://github.com/r-lib/lintr/blob/36236c3c030e64f4a74fb8468d72e8f68b1defbd/R/with.R#L112
